static/js/popovers.js

```diff
@@ -547,6 +547,14 @@ exports.register_click_handlers = function () {
         e.stopPropagation();
     });
 
+    $('body').on('click', '.mark_stream_as_read', function (e) {
+        var stream = $(e.currentTarget).parents('ul').attr('data-name');
+        //var topic = $(e.currentTarget).attr('data-topic-name');
```

Comment:
zerver/models.py has classes defining all the Zulip models, and you can read about the Django query system to get a feel for what you can do, but the basic thing we've been doing is accessing an attributes of another object connected by a foreign key via e.g. `message__recipient` (note the double-`_`)

Topics are just an attribute on Message objects with an index; they don't have their own table.  So for your UserMessage query, you'll want to use `message__topic__iexact=topic` in your filter.  The `iexact` is important since topics in Zulip (like most things) are case-insensitive.